src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,150 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidUserException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String INVALID_DATA = "Invalid user data."
+            + " Please check login, password, and age.";
+    private static final String EXISTS_MSG = "User with this login already exists.";
+    private static RegistrationServiceImpl registrationService;
+
+    @BeforeAll
+    public static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_ValidUserRegistration_NotNullAndEquals() {
+        User user = new User();
+        user.setLogin("testuser1");
+        user.setPassword("password1231");
+        user.setAge(20);
```

Comment:
move user to test class field and set it's fields in the BeforeEach method to avoid code duplicating
```suggestion
    private User user;
    
    @BeforeAll
    public static void setUp() {
        registrationService = new RegistrationServiceImpl();
    }
    
    @BeforeEach
    void setUp() {
        user = new User();
        user.setLogin(VALID_LOGIN);
        user.setPassword(VALID_PASSWORD);
        user.setAge(VALID_AGE);
    }

    @Test
    public void register_ValidUserRegistration_NotNullAndEquals() {
```